### ISSUE TITLE
MPI_F_SYNC_REG is Fortran-only

### ISSUE THREAD

**Issue by dalcinl** [mpiwg-abi/header_and_stub_library#3]
From MPI 4.1, pp 805

_Rationale_. This function is not available in other languages because it would not be useful. This routine has no ierror return argument because there is no operation that can fail. (End of rationale.)

What should we do with this routine here? Remove it from `mpi.h` ? Keep it commented out (and fix the return to `void`) ? Or just remove it?

**Comment by jeffhammond** [comment]
Is it present in any implications now?

**Comment by dalcinl** [comment]
I'm not sure what do you mean. 

1. I'm looking at cleaning up this thing such that MPI implementations can grab the header and use it unmodified, without any stuff that should not be there. 
2. I'm doing some work to make this repository a proper project able to build with make/cmake/meson (to accommodate for other project's needs) such that others can grab it and build against a stub MPI without having to install an actual MPI implementation. Again, for this to work properly the C stubs header and library should not contain alien stuff not corresponding to the C bindings.

**Comment by jeffhammond** [comment]
This symbol does not exist in `mpi.h` of either MPICH or Open MPI so I don't see why we need it.  It belongs in the Fortran modules and nowhere else.
```
% grep MPI_F_SYNC_REG /opt/homebrew/Cellar/mpich/4.1.2/include/mpi.h /opt/homebrew/Cellar/open-mpi/5.0.1/include/mpi.h || echo OK
OK
```

**Comment by jeffhammond** [comment]
To be clear: it should be removed.

**Comment by dalcinl** [comment]
> grep MPI_F_SYNC_REG /opt/homebrew/Cellar/mpich/4.1.2/include/mpi.h /opt/homebrew/Cellar/open-mpi/5.0.1/include/mpi.h

You should have grep with case insensitivity and recursive on the include directory:

```
grep MPI_F_SYNC_REG -ir /opt/homebrew/Cellar/mpich/4.1.2/include /opt/homebrew/Cellar/open-mpi/5.0.1/include
```
but anyway, you would have got the same result. This API is not part of C.


> To be clear: it should be removed.

I'll submit a PR.

PS: If the routine is Fortran-only, I'm wondering why was it called with a `MPI_F_` prefix...

**Comment by dalcinl** [comment]
Fixed via #6.
